RFCs/rfc-0008-wallet-provider-api.md

```diff
@@ -0,0 +1,387 @@
+        scalar: string;
+    };
+};
+
+Provider.request({
+    method: 'mina_signFields',
+    params: { fields: [1, 5, 9, 2] }
+}): Promise<SignedFieldsData>;
+```
+
+#### mina_signTransaction
+Returns a signed transaction.
+
+##### Parameters
+`transaction`: [Required] a well-formed transaction body.
+
+##### Returns
+A signed transaction body.
+
+```typescript
+Provider.request({
+    method: 'mina_signTransaction',
+    params: { transaction: tx}
+}): Promise<SignedTransaction>;
+```
+
+#### mina_sendTransaction
+Submits a signed transaction to the network.
+
+##### Parameters
+- `signedTransaction`: [Required] the signed transaction body
+- `transactionBody`: [Required] the transaction body
+- `transactionType`: [Required] the type of the transaction (e.g. `payment`, `delegation`, `zkapp`)
+
+```typescript
+type TransactionReceipt = {
+    hash: string
+};
+
+Provider.request({
+    method: 'mina_sendTransaction',
+    params: { signedTransaction: signedTx, transactionBody: txBody, transactionType: 'payment'}
+}): Promise<TransactionReceipt>
+```
+#### mina_getBalance
+Returns the balance of the the client's current account.
+
+##### Parameters
+None.
+
+##### Returns
+```typescript
+Provider,request({
+    method: 'mina_getBalance'
+}): Promise<number>
+```
+
+#### mina_createNullifier
+Returns a nullifier.
+
+##### Parameters
+`message`: [Required] an array of `bigint`s
+
+##### Returns
+```typescript
+type Nullifier = {
+    publicKey: Group;
+    public: {
+        nullifier: Group;
+        s: Field;
+    };
+    private: {
+        c: Field;
+        g_r: Group;
+        h_m_pk_r: Group;
+    };
+};
+
+Provider.request({
+    method: 'mina_createNullifier',
+    params: { message: [1n] }
+}): Promise<Nullifier>
+```
+
+
+## Appendix I: Consumer-Facing API Documentation
+
+### request
+
+Makes a Mina RPC method call.
+
+```typescript
+interface RequestArguments {
+  readonly method: string;
+  readonly params?: readonly unknown[] | object;
+}
+
+Provider.request(args: RequestArguments): Promise<unknown>;
+```
+
+The returned Promise resolves with the method's result or rejects with a [`ProviderRpcError`](#errors). For example:
+
+```javascript
+Provider.request({ method: 'mina_accounts' })
+  .then((accounts) => console.log(accounts))
+  .catch((error) => console.error(error));
+```
+
+### Events
+
+Events follow the conventions of the Node.js [`EventEmitter` API](https://nodejs.org/api/events.html).
+
+#### connect
+
+The Provider emits `connect` when it:
+
+- first connects to a chain after being initialized.
+- first connects to a chain, after the `disconnect` event was emitted.
+
+```typescript
+interface ProviderConnectInfo {
+  readonly chainId: string;
+}
+
+Provider.on('connect', listener: (connectInfo: ProviderConnectInfo) => void): Provider;
+```
+
+The event emits an object with a hexadecimal string `chainId` per the `mina_chainId` Mina GraphQL query result, and other properties as determined by the Provider.
```

Comment:
There seems to be no fixed chainId now, and the provider needs to define an enum to enumerate different chainId.

Reply:
An application like a wallet must be aware of the information for the chain it is "connected" to, how it chooses to do that is an implementation detail.

Reply:
What wallets can use to provide more information is something like [this](https://github.com/ChainAgnostic/namespaces/pull/91). In that case, the wallet can provide that standardised `networkID` (e.g. `mina:mainnet`)

Reply:
Yes, we are currently doing this. We define the network ID inside the Auro Wallet to distinguish different networks, because the network ID returned by the node will now change.
```ts
const netType = {
  Mainnet: "mainnet",
  Devnet: "devnet",
  Berkeley: "berkeley",
  Testworld2: "testworld2"
}
```